interpreter/text/parser.mly

```diff
@@ -39,6 +39,37 @@ let ati i =
 let literal f s =
   try f s with Failure _ -> error s.at "constant out of range"
 
+let range_check i32 min max at =
+  let i = Int32.to_int i32 in
+  if i > max || i < min then error at "constant out of range" else i
+
+let int8_of_string s = range_check (I32.of_string s.it) (-128) 255 s.at
```

Comment:
Nit: i8 (same below)
Down the line you'll probably need to introduce proper I8 and I16 modules to implement various lane-wise ops, which would eliminate the range_check hack.

Reply:
Added a TODO for this, can add a I8 and I16 module in a future patch.